metabase/metabase
# Add caching for /api/dataset/native calls and e2e tests

Epic https://github.com/metabase/metabase/issues/26837

How to test:
- New -> SQL query -> `SELECT {{a}}` and set value for the `a` parameter.
- Click `Preview the query` and close the modal.
- Open the modal again. There should be no extra `/api/dataset/native` call.
- Change the query and/or parameter value and open the modal. It should call `/api/dataset/native`.
- New cypress tests should pass


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27113)
